text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,62 @@
+# Definition of maturity levels to be uniformly used by OpenTelemetry SIGs
+
+On 08 Mar 2023, the OpenTelemetry GC and TC held an OpenTelemetry Leadership summit, discussing various topics. One of the themes we discussed was establishing standard rules for describing the maturity of the OpenTelemetry project. This OTEP summarizes what was discussed there and is intended to have the wider community provide feedback.
+
+This OTEP builds on what was previously communicated by the project, especially the [Versioning and stability for OpenTelemetry clients](https://opentelemetry.io/docs/reference/specification/versioning-and-stability).
+
+The Collector's [stability levels](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) inspired the maturity levels.
+
+## Motivation
+
+Quite often, the community is faced with the question of the quality and maturity expectations of its diverse set of components. This OTEP aims to bring clarity by establishing a framework to communicate the maturity for components delivered by SIGs in the name of the project. As the OpenTelemetry project comprises a multitude of SIGs, and each SIG has several components of varying quality, having this framework will help set the right expectations for OpenTelemetry users using a unified nomenclature.
+
+## Explanation
+
+### Maturity levels
+
+Deliverables of a SIG MUST have a declared maturity level, established by SIG maintainers (SIGs), likely with the input of the code owners. The maturity level of a deliverable is, at most, the lowest level of the user-visible dependencies. For instance, if the Collector's component "otlpreceiver" declares a dependency on the OpenTelemetry Collector API "config" package which is marked with a maturity level of "beta", the "otlpreceiver" can be at most "beta". The OpenTelemetry Collector "core" distribution is also affected so that it can be at most "beta". Only once all dependencies are marked as "stable" MAY a component be marked as "stable".
+
+#### Development
+
+Not all pieces of the component are in place yet, and it might not be available for users yet. Bugs and performance issues are expected to be reported. User feedback is desired, especially regarding the user experience (configuration options, component observability, technical implementation details, ...). Configuration options might break often depending on how things evolve. The component SHOULD NOT be used in production. The component MAY be removed without prior notice.
```

Comment:
>User feedback is desired, especially regarding the user experience

Do we want this at this this level? I wonder if that may be premature. Should we instead start recommending this from "Alpha" level?

Reply:
As a developer of a component, I would appreciate to hear about possible use-cases for my component. I wouldn't care too much about performance issues, though.

Reply:
@tigrannajaryan, do you think my points are reasonable, or do you think the text needs a change?

Reply:
I prefer to avoid actively soliciting feedback for a "Development" component that is just introduced. If I have just created it, and it likely is half-broken (as expected for this level) I don't want to be receiving comments from users saying it is broken. I know it already. I understand we are seeking feedback because it may be valuable, but I suspect signal-to-noise ratio is going to be low. 

I am not strongly opposed to keeping the text as is if that's what the majority of the community prefers, but would like to see a few voices in favour of it.

Reply:
How about rewording to something like this:

> User feedback around the UX of the component is desired, such as for configuration options, component observability, technical implementation details, and planned use-cases for the component.

Reply:
Sounds good.